corehq/apps/app_manager/decorators.py

```diff
@@ -66,13 +69,27 @@ def _safe_cached_download(request, *args, **kwargs):
         # make endpoints that call the user fail hard
         from django.contrib.auth.models import AnonymousUser
         request.user = AnonymousUser()
+        username = request.GET.get('username')
         if request.GET.get('username'):
             request.GET = request.GET.copy()
             request.GET.pop('username')
 
         latest_enabled_build = None
-        if latest and request.GET.get('profile') and toggles.RELEASE_BUILDS_PER_PROFILE.enabled(domain):
-            latest_enabled_build = get_latest_enabled_build_for_profile(domain, request.GET.get('profile'))
+        if latest and toggles.MANAGE_RELEASES_PER_LOCATION.enabled(domain):
```

Comment:
There's a lot of code here about how to get the correct app.  I'd pull it out to a standalone function so it's easier to understand.  That'd also be a good place to test.